pennylane/devices/default_clifford.py

```diff
@@ -460,57 +519,239 @@ def simulate(
+                        0
+                    ][0]
+                )
+            # Computing classical shadows via manual single sampling
+            elif isinstance(meas, ClassicalShadowMP):
+                results.append(self._measure_classical_shadow(stim_circuit, circuit, meas, stim))
 
-                # Computing statevector via tableaus
-                elif isinstance(meas, StateMP):
-                    res = self._measure_state(tableau_simulator, circuit, global_phase)
+            # Computing observable expectation value using above classical shadows
+            elif isinstance(meas, ShadowExpvalMP):
+                results.append(self._measure_expval_shadow(stim_circuit, circuit, meas, stim))
 
-                # Computing purity via tableaus
-                elif isinstance(meas, PurityMP):
-                    res = self._measure_purity(meas, circuit)
+            elif isinstance(meas, ExpectationMP):
+                results.append(
+                    self._measure_expectation_samples(stim_circuit, meas_op, num_shots, sample_seed)
+                )
 
-                # Computing expectation values via measurement
-                elif isinstance(meas, ExpectationMP):
-                    res = self._measure_expectation(tableau_simulator, meas, stim)
+            elif isinstance(meas, VarianceMP):
+                meas_obs = qml.operation.convert_to_opmath(meas_op)
+                with qml.operation.use_new_opmath():
+                    meas_obs1 = meas_obs.simplify()
+                    meas_obs2 = (meas_obs1**2).simplify()
 
-                # Computing more measurements
-                else:
-                    raise NotImplementedError(
-                        f"default.clifford doesn't support the {type(meas)} measurement at the moment."
+                # use the naive formula for variance, i.e., Var(Q) = ⟨𝑄^2⟩−⟨𝑄⟩^2
+                vars = (
+                    self._measure_expectation_samples(
+                        stim_circuit, meas_obs2, num_shots, sample_seed
+                    )
+                    - self._measure_expectation_samples(
+                        stim_circuit, meas_obs1, num_shots, sample_seed
                     )
+                    ** 2
+                )
+                results.append(vars)
+
+            # Computing rest of the measurement by processing samples
+            else:
+                samples = qml.math.array(
+                    [
+                        self._measure_observable_sample(
+                            meas_op, stim_circuit, num_shots, sample_seed
+                        )[0]
+                    ]
+                )
+                wire_order = {wire: idx for idx, wire in enumerate(meas.wires)}
+                results.append(meas.process_samples(samples=samples, wire_order=wire_order))
+        return results
+
+    def measure_analytical(self, circuit, stim_circuit, tableau_simulator, global_phase, stim):
```

Comment:
Instead of having to pass around the package as a keyword argument, maybe we could follow how we handle importing matplotlib.  In `MPLDrawer.py`, we import matplotlib globally at the top, but defer raising an import error until we try and initialize the class: 

https://github.com/PennyLaneAI/pennylane/blob/321ad706c219ad689de4ee163cf8c37ef017e081/pennylane/drawer/mpldrawer.py#L22

That way we can still have the lazy error, but the package can be global as expected.